File: app/src/main/java/com/lucianos/rentacar/data/FlowState.kt
package com.lucianos.rentacar.data

object EntregaState {
    var reservationId: String = ""
    var docLic: Boolean = false
    var docIne: Boolean = false
    var docDom: Boolean = false
    var docTarj: Boolean = false
    var signed: Boolean = false
    var damageCount: Int = 0
    var photoCount: Int = 0
    var fuelLevel: String = "3/4"
    var kmStart: String = "0"
    var paymentMethod: String = "cash"
    var amountReceived: String = ""

    fun reset(resId: String) {
        reservationId = resId
        docLic = false; docIne = false; docDom = false; docTarj = false
        signed = false; damageCount = 0; photoCount = 0
        fuelLevel = "3/4"; kmStart = "0"
        paymentMethod = "cash"; amountReceived = ""
    }

    val allDocsOk get() = docLic && docIne && docDom && docTarj
}

object DevolucionState {
    var reservationId: String = ""
    var kmReturn: String = "0"
    var fuelLevel: String = "1/2"
    var newDamageCount: Int = 0
    var returnMethod: String = "cash"
    var photoTaken: Boolean = false
    var signed: Boolean = false

    fun reset(resId: String) {
        reservationId = resId
        kmReturn = "0"; fuelLevel = "1/2"; newDamageCount = 0
        returnMethod = "cash"; photoTaken = false; signed = false
    }

    val kmExtra: Int get() {
        val km = kmReturn.toIntOrNull() ?: 0
        val recorrido = km - 45200
        return maxOf(0, recorrido - 600)
    }
    val fuelDeficit: Int get() = mapOf("E" to 4, "1/4" to 3, "1/2" to 2, "3/4" to 0, "F" to 0)[fuelLevel] ?: 0
    val extraCharges: Int get() = kmExtra * 3 + fuelDeficit * 80 + newDamageCount * 800
    val depositAmount: Int get() = 2000
    val toReturn: Int get() = maxOf(0, depositAmount - extraCharges)
    val toCharge: Int get() = maxOf(0, extraCharges - depositAmount)
}

File: web/src/state/flowState.ts
export const entregaState = {
  reservationId: '',
  docLic: false, docIne: false, docDom: false, docTarj: false,
  signed: false, damageCount: 0, photoCount: 0,
  fuelLevel: '3/4', kmStart: '0',
  paymentMethod: 'cash', amountReceived: '',
  get allDocsOk() { return this.docLic && this.docIne && this.docDom && this.docTarj },
  reset(resId: string) {
    this.reservationId = resId
    this.docLic = false; this.docIne = false; this.docDom = false; this.docTarj = false
    this.signed = false; this.damageCount = 0; this.photoCount = 0
    this.fuelLevel = '3/4'; this.kmStart = '0'
    this.paymentMethod = 'cash'; this.amountReceived = ''
  },
}

export const devolucionState = {
  reservationId: '',
  kmReturn: '0', fuelLevel: '1/2', newDamageCount: 0,
  returnMethod: 'cash', photoTaken: false, signed: false,
  get kmExtra() {
    const km = parseInt(this.kmReturn) || 0
    const recorrido = km - 45200
    return Math.max(0, recorrido - 600)
  },
  get fuelDeficit() {
    const map: Record<string, number> = { E: 4, '1/4': 3, '1/2': 2, '3/4': 0, F: 0 }
    return map[this.fuelLevel] ?? 0
  },
  get extraCharges() { return this.kmExtra * 3 + this.fuelDeficit * 80 + this.newDamageCount * 800 },
  get depositAmount() { return 2000 },
  get toReturn() { return Math.max(0, this.depositAmount - this.extraCharges) },
  get toCharge() { return Math.max(0, this.extraCharges - this.depositAmount) },
  reset(resId: string) {
    this.reservationId = resId
    this.kmReturn = '0'; this.fuelLevel = '1/2'; this.newDamageCount = 0
    this.returnMethod = 'cash'; this.photoTaken = false; this.signed = false
  },
}
